### ISSUE TITLE
Provide SvelteKit version next to each feature

### ISSUE THREAD

**Issue by theetrain** [sveltejs/kit#11358]
### Describe the problem

The documentation represents the latest version of SvelteKit. Users who are behind major or minor versions may not be able to use certain features; or a module's API or behaviours changed; and there's currently no easy indicator on the docs to guide them.

### Describe the proposed solution

- Leverage [JSDoc @since](https://jsdoc.app/tags-since) to mark when a module or feature was last-updated; and render that version on the docs. 
- Manually indicate which features were available since the latest compatible SvelteKit version.

### Alternatives considered

Look at historic docs through git, or comb the changelog.

### Importance

nice to have

### Additional Information

Tangentially related to https://github.com/sveltejs/kit/issues/6023

**Comment by benmccann** [comment]
For 2.0 we did try to call out differences between 1.x and 2.x inline in the docs. If there are any we missed please let us know so that we can address them!

I like the idea of using `@since` moving forward as well

**Comment by theetrain** [comment]
> For 2.0 we did try to call out differences between 1.x and 2.x inline in the docs

I noticed some of those comments, such as the callout in Streaming with Promises; glad to see them! If I see anything that can use such clarification, I'll to try to contribute to them case-by-case.

As for this ticket, I'm hopeful to see the Kit (and possibly Svelte) sites be able to interpret `@since`. Once decided, I can help contribute to that as well. 

This would particularly be handy for less proactive users who may discover a v2.1.0 feature months later. Having that version highlighted next to the feature in the docs would help a lot before they find out it doesn't work in their installed version of SvelteKit.

**Comment by benmccann** [comment]
I added some `@since` tags here: https://github.com/sveltejs/kit/tree/since

I can't build the site currently though on `master`. Perhaps @PuruVJ could take a look at that?

**Comment by benmccann** [comment]
I sent https://github.com/sveltejs/kit/pull/11569 to add `@since` tags to the types. They wouldn't be surfaced in the site yet, but hopefully it help folks in VS Code and it'll get the content there if someone wants to update the site. @PuruVJ it could be something you might be interested in

**Comment by igorsantos07** [comment]
>I'm sorry for the rude tone below, but I cannot find a better way to convey my astonishment.

I'm mesmerized to realize there are **no** v1 docs. I understand there were little changes, but even then, a callout of sorts to say "hey, the example you just saw above won't work if you didn't update yet, do as follows.." is just......... counterproductive. I don't even have a proper adjective to that. It makes no sense reading about the new version syntax and _then_ rewiring your understanding after checking the callout below.

_Almost_ other language, library or framework out there have a select at the top or bottom for you to indicate which version you want to read about. Those who don't (I can only think of PHP) have version descriptions at the title of the function you're reading about. But even then, in some scenarios where it's a matter of syntax (e.g. throwing or not `redirect` or `error`), integrating `@since` with some sort of version heading won't be enough - different versions _need_ different docs in these scenarios.

I should add that I only discovered there were _little changes_ after googling a bit about the versions. Once I realized I was reading v2 while working with v1, I instantly freaked out and stopped checking the docs, since you never know _what_ changed. Another idea is to leave a banner on the top of all doc pages, stating "this is v2 of docs; click here to see what changed from v1", so you at least can piece stuff together.

Other than that, I urge you all to find a better way to compile v1 docs and make it live side by side with v2. The current way makes me think you're just forcing us to upgrade, by getting rid of v1 docs.

Lastly, I must say part of this discontentment is probably related to npm as well - it refused to install `adapter-static` because it doesn't seem able to realize I need v2 and not v3 of that package :joy:
